src/main/java/uk/gov/justice/hmpps/prison/service/PrisonerProfileUpdateService.kt

```diff
@@ -44,7 +51,61 @@ class PrisonerProfileUpdateService(
     }
   }
 
-  private fun Optional<Offender>.orElseThrowNotFound(message: String, prisonerNumber: String) =
+  @Transactional
+  fun updateNationalityOfCurrentBooking(prisonerNumber: String, nationality: String?) {
+    val profileType: ProfileType = profileTypeRepository.nationalityProfile().getOrThrow()
+    val profileCode: ProfileCode? =
+      nationality?.uppercase()?.let { profileCodeRepository.profile(profileType, it).getOrThrow() }
+
+    try {
+      val currentBooking = currentBooking(prisonerNumber)
```

Comment:
Meh.. I think 'latest' over 'current' would be preferred since if the prisoner has been released, the latest booking is inactive